core/src/main/java/bio/terra/pearl/core/dao/workflow/ParticipantTaskDao.java

```diff
@@ -50,6 +50,27 @@ public Map<UUID, List<ParticipantTask>> findByEnrolleeIds(Collection<UUID> enrol
                 .stream().collect(Collectors.groupingBy(ParticipantTask::getEnrolleeId, Collectors.toList()));
     }
 
+    public Map<UUID, ParticipantTask> findLatestNotRemovedByEnrolleeIds(Collection<UUID> enrolleeIds, String stableId) {
+        if (enrolleeIds.isEmpty()) {
+            return Collections.emptyMap();
+        }
+
+        return jdbi.withHandle(handle ->
+                handle.createQuery("""
+                                select * from %s
+                                where enrollee_id in (<enrolleeIds>)
+                                and status not in ('REMOVED', 'REJECTED')
+                                and target_stable_id = :stableId
+                                order by created_at desc
+                                """.formatted(tableName))
```

Comment:
```suggestion
                                select distinct on (enrollee_id, target_stable_id) * from %s
                                where enrollee_id in (<enrolleeIds>)
                                and status not in ('REMOVED', 'REJECTED')
                                and target_stable_id = :stableId
                                order by enrollee_id, target_stable_id, created_at desc
                                """.formatted(tableName))
```
will only return the rows you want -- there might be a lot of tasks for studies with a lot of recurrence like A-T